core/src/apps/bitcoin/verification.py

```diff
@@ -0,0 +1,93 @@
+from trezor import wire
+from trezor.crypto.hashlib import sha256
+
+from apps.common.coininfo import CoinInfo
+
+from .common import ecdsa_hash_pubkey, ecdsa_verify
+from .scripts import (
+    input_script_p2wpkh_in_p2sh,
+    input_script_p2wsh_in_p2sh,
+    output_script_native_p2wpkh_or_p2wsh,
+    output_script_p2pkh,
+    output_script_p2sh,
+    read_input_script_multisig,
+    read_input_script_p2pkh,
+    read_output_script_multisig,
+    read_witness_p2wpkh,
+    read_witness_p2wsh,
+)
+
+
+class SignatureVerifier:
+    def __init__(
+        self, script_pubkey: bytes, script_sig: bytes, witness: bytes, coin: CoinInfo,
+    ):
+        self.threshold = 1
```

Comment:
please declare the member variables and their types here as well, the code structure makes it rather difficult to find them :)

Reply:
c5a26ce6211dc170d01d33db41d099445d837101